src/OrbitGl/ScopedStatus.h

```diff
@@ -67,10 +67,10 @@ class ScopedStatus final {
     if (std::this_thread::get_id() == data_->main_thread_id) {
       data_->status_listener->UpdateStatus(data_->status_id, message);
     } else {
-      data_->main_thread_executor->Schedule(
-          [status_id = data_->status_id, status_listener = data_->status_listener, message] {
-            status_listener->UpdateStatus(status_id, message);
-          });
+      auto executor = data_->main_thread_executor.lock();
+      if (executor == nullptr) return;
+      executor->Schedule([status_id = data_->status_id, status_listener = data_->status_listener,
+                          message] { status_listener->UpdateStatus(status_id, message); });
```

Comment:
Looking at this I think it might be a good idea to introduce a function like this:
template <typename F>
bool TrySchedule(std::weak_ptr<MainThreadExecutor> executor, F&& functor);

Reply:
Yes, I like the idea. On it. And I would make it nodiscard, WDYT?

Reply:
```
template <typename F>
std::optional<orbit_base::Future<T>> TrySchedule(std::weak_ptr<MainThreadExecutor> executor, F&& functor);
```

The resulting type should probably be an optional future. And I take the nodiscard back.

Reply:
makes sense.

Reply:
I added the function, including tests in MainThreadExecutorImplTest.cpp